### PR TITLE
Simple Component Revisions

### PR DESCRIPTION
**Pull request by MichaelChung123**
# Description
Making revision changes to all of the Simple Custom Components in the Form IO Builder.

The list of changes can be found in the ticket description.
https://apps.nrs.gov.bc.ca/int/jira/secure/RapidBoard.jspa?selectedIssue=SHOWCASE-1898&rapidView=2025&view=detail&projectKey=SHOWCASE

## Types of changes
Made changes to our existing Form IO Custom Components.

## Checklist

- [X] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [X] I have checked that unit tests pass locally with my changes
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
PR Deployed at: https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-203

File: app/frontend/src/assets/scss/style.scss [hunk 1]
```diff
@@ -361,6 +361,19 @@ a,
...
+  // Removing default (optional) flag on Custom Day Component
+  label[for$="simpleday"]:after {
+    content: "";
+  }
+
+  // Adding (optional) flag to Checkbox cases
+  :is(.form-check.checkbox) label {
+      &:after {
+        content: " (optional)";
+        color: inherit;
```
Comment (TimCsaky) [inline app/frontend/src/assets/scss/style.scss:373]: here you are overriding line 353. so you could just change line 353 to:
```
label {
```

File: app/frontend/src/assets/scss/style.scss [hunk 2]
```diff
@@ -361,6 +361,19 @@ a,
...
+      }
+  }
```
Comment (TimCsaky) [inline app/frontend/src/assets/scss/style.scss:375]: here you are overriding line 353. so you could just change line 353 to:
` label { ` and you wont need this

Comment (MichaelChung123) [reply app/frontend/src/assets/scss/style.scss:375]: I was actually writing this in order to override line 353.

1. Line 365 is used to target the label for the Day Component
2. Line 370 is targeting the .form-check.checkbox class to remove the optional flag by overriding Line 353

We can talk about scss refactor on Friday

Comment (MichaelChung123) [reply app/frontend/src/assets/scss/style.scss]: After clarification, this has been changed and fixed.

**Comment by MichaelChung123** [comment]
Rebased

**Comment by bcgov-nr-csst** [comment]
PR Deployed at: https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-203

**Review by loneil — APPROVED** [review]
I think this is probably ok, tried it out on one complex form (parks complaint) that I have. **Lots** of components changed though so we should do some solid regression on existing forms in Test before releasing.

**Review by loneil — COMMENTED** [review]
Actually might be 2 code climate fixes that can be addressed (the duplicate ones in components will have to be ignored)

image

File: components/src/components/index.ts [hunk 3]
```diff
@@ -38,17 +34,13 @@ export default {
...
   simplenumber,
   simplepanel,
-  simpleparagraph,
   simplephonenumber,
   simpleradios,
   simpleselect,
   simpletabs,
   simpletextarea,
   simpletextfield,
-  simpletime
```
Comment (jujaga) [inline components/src/components/index.ts:53]: I'm a bit concerned about removing these components from the export package - have we verified that forms using these components already do not break?

File: components/src/components/SimpleEmail/editForm/Component.edit.data.ts [hunk 4]
```diff
@@ -1,35 +1,25 @@
...
+      input: true,
+      values: [
+          {
+              value: 'mixed',
+              label: 'Mixed (Allow upper and lower case)'
+          },
+          {
+              value: 'uppercase',
+              label: 'Uppercase'
+          },{
```
[the remark this answers is not in the document]
Comment (MichaelChung123) [reply components/src/components/SimpleEmail/editForm/Component.edit.data.ts:19]: good catch haha

File: components/src/components/OrgBook/editForm/Orgbook.edit.display.ts [hunk 5]
```diff
@@ -1 +1,5 @@
-export default [];
+
```
Comment (jujaga) [inline components/src/components/OrgBook/editForm/Orgbook.edit.display.ts:1]: Extra whitespace line.

File: components/src/components/SimpleColumns4/editForm/Component.edit.display.ts [hunk 6]
```diff
@@ -1,4 +1,7 @@
+
```
Comment (jujaga) [inline components/src/components/SimpleColumns4/editForm/Component.edit.display.ts:1]: Extra whitespace line

File: components/src/components/SimpleEmail/editForm/Component.edit.data.ts [hunk 7]
```diff
@@ -1,35 +1,26 @@
...
+          },
+          {
+              value: 'uppercase',
+              label: 'Uppercase'
+          },
+          {
+              value: 'lowercase',
+              label: 'Lowercase'
+          }
+      ]
```
Comment (jujaga) [inline components/src/components/SimpleEmail/editForm/Component.edit.data.ts:24]: Indentation here looks a bit off. I think this was one of the files that was originally with 4 spaces that you are reformatting to have 2 spaces instead.

**Review by jujaga — APPROVED** [review]
Ran this through the upcoming kitchen-sink cypress test and it looks like it passed. Tentatively approving this with minor style changes requested. 👍

**Review by TimCsaky — CHANGES_REQUESTED** [review]
huh.. a lot of changes here.
as i mentioned, i think the ticket says to include the 'advanced conditions' dropdown on the conditional tab for all simple components. other than that, i believe you got everything!

**Comment by MichaelChung123** [comment]
Rebased

File: components/src/components/Common/Evaluator.ts [hunk 8]
```diff
@@ -0,0 +1,7 @@
+import _ from 'lodash';
```
Comment (jujaga) [inline components/src/components/Common/Evaluator.ts:1]: We're not using lodash in this file at all.

**Review by TimCsaky — APPROVED** [review]
confirmed the advanced conditional stuff was working.
all looking good!

File: components/src/components/Common/Evaluator.ts (hunk 8 above)
Comment (MichaelChung123) [reply components/src/components/Common/Evaluator.ts]: removed in latest commit

**Pull request merged**